### PR TITLE
feat: Add generic data loader

### PR DESCRIPTION
**Pull request by ronkatz96**

### CODE REVIEW

File: packages/polaris-core/src/data-loaders/data-loader-service.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+    public initDataLoader = (realityId: number, className: any) =>
+        new DataLoader<string, any>(
+            async (ids: readonly string[]): Promise<any> => {
+                const entities: any[] = await getConnectionForReality(
+                    realityId,
+                    this.supportedRealities as any,
+                    this.connectionManager as PolarisConnectionManager,
+                )
+                    .getRepository<typeof className>(className.constructor.name)
```
Comment (chenshoo) [inline packages/polaris-core/src/data-loaders/data-loader-service.ts]: you should pass context here.

File: packages/polaris-core/src/server/configurations-manager.ts [hunk 2]
```diff
@@ -89,5 +90,9 @@ export const getPolarisServerConfigFromOptions = (
         permissionsConfig: options.permissionsConfig || getDefaultPermissionsConfiguration(),
         enableDataVersionFilter:
             options.enableDataVersionFilter === undefined ? true : options.enableDataVersionFilter,
+        dataLoaderService: new DataLoaderService(
```
Comment (chenshoo) [inline packages/polaris-core/src/server/configurations-manager.ts]: initialize it after you have context

File: packages/polaris-core/src/utils/data-loader-util.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import DataLoader = require('dataloader');
+
+export const getDataLoader = (
+    entityType: string,
+    context: PolarisGraphQLContext,
+    className: any,
+) => {
+    const realityId = context.reality.id;
+    const dataLoader = context.dataLoaders?.find(
+        (value) => value.realityId === realityId && value.entityType === entityType,
```
Comment (chenshoo) [inline packages/polaris-core/src/utils/data-loader-util.ts]: I think the reality id is redundant cause you only have one reality per request

File: packages/polaris-test/tests/date-filter.test.ts [hunk 4]
```diff
@@ -125,15 +125,15 @@ describe('date filter tests', () => {
                 await polarisTest(server, async () => {
                     await graphQLRequest(createBook.request, {}, { title: 'book1' });
                     await graphQLRequest(createBook.request, {}, { title: 'book2' });
```
Comment (chenshoo) [inline packages/polaris-test/tests/date-filter.test.ts:127]: undelete the test.

File: packages/polaris-core/guides/graphql/dataloader/README.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        createdBy: String!
+        creationTime: DateTime!
+        lastUpdatedBy: String
+        lastUpdateTime: DateTime
+        realityId: Int!
+        title: String
+        author: Author
+        chapters: [Chapter]
+        reviews: [Review]
+    }
```
Comment (osher-sade) [inline packages/polaris-core/guides/graphql/dataloader/README.md]: prettify

File: packages/polaris-core/guides/graphql/dataloader/README.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+This is how chapters are defined in `Book`
+```typescript
+@OneToMany(() => Chapter, (chapters) => chapters.book)
+    public chapters: Chapter[];
+```
+
+This is how book is defined in `Chapter`
+```typescript
+@ManyToOne(() => Book, (book) => book.chapters, { onDelete: 'CASCADE' })
+    public book: Book;
```
Comment (osher-sade) [inline packages/polaris-core/guides/graphql/dataloader/README.md]: prettify

File: packages/polaris-test/tests/jsonRequestsAndHeaders/allBooksWithChaptersDataFetcher.json [hunk 7]
```diff
@@ -0,0 +1,4 @@
+{
+  "request": "{\n  allBooks {\n    id\n    title\n       chapters {\n      id\n      }\n  }\n}"
+
+}
```
Comment (osher-sade) [inline packages/polaris-test/tests/jsonRequestsAndHeaders/allBooksWithChaptersDataFetcher.json]: insert new line at the end of this file

bruh

**Review by osher-sade — APPROVED** [review]

**Pull request merged**